### PR TITLE
nspr: bump version

### PR DESCRIPTION
**Pull request by kenmays**
Bumps to NSPR 4.26. Built and tested on Haiku hrev54390 x86_64.
Affects: calligra, libreoffice, xmlsec, okular, qca, nss

### CODE REVIEW

**Comment by Begasus** [comment]
Please add a list on depending recipes that needs updating, and prepare required version bump/version updates for the recipes using this, for instance:
```
[ ] nss
    [ ] xmlsec
    [ ] qca
[ ] libreoffice ...

**Comment by Begasus** [comment]
On that note, version 4.26 worked OK for me building nss, also check gcc2h, as some of the recipes require fixes for conflicting int(*) types

**Comment by kenmays** [comment]
x86_64 is good now. On x86, my NSS 3.46.1-1 x86 package was installed on Nov 11, 2019, so it built fine back then on x86.

File: dev-libs/nspr/patches/nspr-4.26.patchset [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+index b62f1cd..0687af4 100644
+--- a/configure.in
++++ b/configure.in
+@@ -1483,6 +1483,30 @@ tools are selected during the Xcode/Developer Tools installation.])
+     PR_MD_CSRCS=freebsd.c
+     ;;
+ 
++*-haiku*)
++    AC_DEFINE(XP_UNIX)
++    AC_DEFINE(XP_HAIKU)
```
Comment (Begasus) [inline dev-libs/nspr/patches/nspr-4.26.patchset:24]: I think we need to remove this line (and adjust references to it), this was only added after nspr version 4.19, prior to that it didn't exists, using !defined(__HAIKU__) for the PRUint(32) types fixed the build for xmlsec ...
```diff
- +    AC_DEFINE(XP_HAIKU)
```

File: dev-libs/nspr/patches/nspr-4.26.patchset [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+diff --git a/pr/include/obsolete/protypes.h b/pr/include/obsolete/protypes.h
+index 5400afb..0a21cbf 100644
+--- a/pr/include/obsolete/protypes.h
++++ b/pr/include/obsolete/protypes.h
+@@ -9,6 +9,9 @@
+  * possible time. The NSPR API is implemented and documented using
+  * the new definitions.
+  */
++#if defined(XP_HAIKU)
++#include <support/SupportDefs.h>
```
Comment (Begasus) [inline dev-libs/nspr/patches/nspr-4.26.patchset:173]: #include <SupportDefs.h> should be enough here

**Comment by Begasus** [comment]
@kenmays could you check https://paste.ubuntu.com/p/SnTPQyKMSj/ ?
It fixed the build for me on gcc2h (works ok with nss and xmlsec)

File: dev-libs/nspr/patches/nspr-4.26.patchset (hunk 1 above)
Comment (kenmays) [reply dev-libs/nspr/patches/nspr-4.26.patchset]: Done

**Comment by Begasus** [comment]
Could you leave out the other packages and just add the latest one here and for nss, would be a bad idea if things are not working out on the buildbot and things would have to be reverted again, if things are ok the older could still be removed in a seperate PR

**Comment by Begasus** [comment]
Why close this? we've put already quite some work into it

**Comment by kenmays** [comment]
Oh...

File: dev-libs/nspr/nspr-4.26.recipe [hunk 3]
```diff
@@ -4,21 +4,26 @@ Netscape Portable Runtime (NSPR) provides a platform-neutral API for system \
...
-ARCHITECTURES="!x86_gcc2 x86 x86_64"
-SECONDARY_ARCHITECTURES="x86"
+SOURCE_URI="https://ftp.mozilla.org/pub/mozilla.org/nspr/releases/v$portVersion/src/nspr-$portVersion.tar.gz"
+CHECKSUM_SHA256="fc9d142d85b74ffd2e6374a0c9016f3f2dac074225e24df3070e5a72d31b773d"
 SOURCE_DIR="nspr-$portVersion/nspr"
+if [ $effectiveTargetArchitecture == 'x86_64' ]; then
+		PATCHES="nspr-$portVersion.patchset"
+else
+		PATCHES="nspr_x86-$portVersion.patchset"
+fi
```
Comment (Begasus) [inline dev-libs/nspr/nspr-4.26.recipe:22]: At first look I don't see any difference in both patchsets, if so, only 1 should be needed

File: dev-libs/nspr/patches/nspr-4.26.patchset [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
++#if !defined(__HAIKU__)
+ typedef PRUint64 uint64;
++#endif
+ 
+ /*
+  * uint32
+  */
+ 
+-#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO)
++#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO) && defined(__HAIKU__)
```
Comment (Begasus) [inline dev-libs/nspr/patches/nspr-4.26.patchset:209]: This is so far as I found the only changed line, but this should be the same as it is used on the other patchset

**Comment by Begasus** [comment]
DESCRIPTION should be re-written (starting on the same line ...)
LICENSE should use tabs for multiple lines

**Comment by Begasus** [comment]
This needs to be applied in the patchset also: https://github.com/haikuports/haikuports/blob/69d9f8fe3fa2718ec1296074c5947efaa387bac8/dev-libs/nspr/patches/nspr-4.19.patchset#L280

File: dev-libs/nspr/nspr-4.26.recipe (hunk 3 above)
Comment (Begasus) [reply dev-libs/nspr/nspr-4.26.recipe:19]: No need to use seperate patchsets here ...

File: dev-libs/nspr/patches/nspr-4.26.patchset (hunk 4 above)
Comment (Begasus) [reply dev-libs/nspr/patches/nspr-4.26.patchset:209]: This should be changed to reflect the one from the _x86 patchset:
``diff
-+#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO) && defined(__HAIKU__)
++#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO) && !defined(__HAIKU__)
```

File: dev-libs/nspr/nspr-4.26.recipe (hunk 3 above)
Comment (kenmays) [reply dev-libs/nspr/nspr-4.26.recipe]: Fixed

File: dev-libs/nspr/nspr-4.26.recipe [hunk 5]
```diff
@@ -1,23 +1,20 @@
...
 HOMEPAGE="https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSPR"
 COPYRIGHT="Mozilla Foundation"
-LICENSE="
-GNU GPL v2
-GNU LGPL v2.1
-MPL v2.0
-"
+LICENSE="GNU GPL v2
+	GNU LGPL v2.1
+	MPL v2.0"
```
Comment (Begasus) [inline dev-libs/nspr/nspr-4.26.recipe:10]: Double qoute moves to a new line here :)
```diff
- 	MPL v2.0"
+ 	MPL v2.0
+ 	"
```

**Pull request merged**